spec/models/searchgov_url_spec.rb

```diff
@@ -890,6 +890,75 @@
       end
     end
 
+    context 'when url has different canonical_url' do
+      let(:canonical_url) { 'https://www.agency.gov/new.html' }
+
+      before do
+        expect(I14yDocument).not_to receive(:create).
+          with(hash_including(title: 'My Title', description: 'My description'))
+        stub_request(:get, url).
+          to_return(status: 301,
+                    body: html,
+                    headers: { location: canonical_url })
+        stub_request(:get, canonical_url).
+          to_return(status: 200,
+                    body: html,
+                    headers: { content_type: 'text/html' })
+      end
+
+      it 'creates a new url' do
+        expect(described_class).to receive(:create).with(url: canonical_url)
+        fetch
+      end
```

Comment:
We could also add in an additional test to make sure the original/non-canonical url is not fetched:

```
it 'does not fetch the original url' do
     fetch
     expect(described_class).not_to have_received(:create).with(url: url)
end
```